ckanext/dataset_subscriptions/actions/__init__.py

```diff
@@ -25,6 +25,9 @@ def latin_username_send_notification(user, email_dict):
 
 
 def dms_notification_provider(user_dict, since):
+    # check if email notifications are enabled for this user, otherwise return empty
```

Comment:
Redundant comment, especially since it explains what the code does not why. If you really think that `user_dict.get('activity_streams_email_notifications')` is not clear enough (I would say it's ok) you can create a local variable:
```
email_notifications_enabled = user_dict.get('activity_streams_email_notifications')
if not email_notifications_enabled:
  return []
```
This way your code is "self explaining" and much easier to maintain. Comments are very easy to miss during updates and you end up with misleading info in comments which IMO is worse then never having any comment at all.